src/recoil/selectors.ts

```diff
@@ -0,0 +1,87 @@
+import { DefaultValue, selector } from 'recoil';
+import { itemDetailsState, itemsState } from './atoms';
+import { Products } from '../types/Product';
+import { UpdateLocalStorage } from '../utils/UpdateLocalStorage';
+import { fetchProducts } from '../api';
+
+/**
+ * 전체 금액, 배송비 계산, 총 결제 금액 계산
+ */
+export const totalPriceSelector = selector({
+  key: 'totalPriceSelector',
+  get: ({ get }) => {
+    const productIds = get(itemsState);
+    let totalAmount = 0;
+    productIds.forEach((itemsState) => {
+      const { quantity, price, isChecked } = get(
+        itemDetailsState(itemsState.id),
+      );
+      if (isChecked) {
+        totalAmount += quantity * price;
+      }
+    });
+    const deliveryFee = totalAmount >= 100000 || totalAmount === 0 ? 0 : 3000;
+    const calculatedTotalAmount = totalAmount + deliveryFee;
+    return { totalAmount, deliveryFee, calculatedTotalAmount };
+  },
+});
+
+/**
+ * get: () => boolean
+ * set: (newValue: 변경할 boolean 값) => void
+ * 전체 선택 체크 시 모든 itemDetailsState의 isChecked 변경,
+ * LocalStorage 업데이트
+ */
+export const toggleAllSelector = selector<boolean>({
+  key: 'toggleAllSelector',
+  get: ({ get }): boolean => {
+    const items: Products[] = get(itemsState);
```

Comment:
이 부분은 별도로 타입을 설정하지 않아도 타입 추론이 자동적으로 되는 것 같아요!

```suggestion
    const items = get(itemsState);
```